File: Leetcode-Problems/Find Words That Can Be Formed By Characters.cpp
class Solution
{
public:
    bool check(string s, int arr[])
    {
        int temp[26] = {0};
        for (char c : s)
        {
            temp[c - 'a']++;
        }
        for (int i = 0; i < 26; i++)
        {
            if (temp[i] > arr[i])
                return false;
        }
        return true;
    }

    int countCharacters(vector<string> &words, string chars)
    {
        int arr[26] = {0};
        for (char c : chars)
        {
            arr[c - 'a']++;
        }
        int ans = 0;
        for (string ch : words)
        {
            if (check(ch, arr))
            {
                ans += ch.length();
            }
        }
        return ans;
    }
};

File: Leetcode-Problems/Find Words That Can Be Formed By Characters.java
class Solution {
    public int countCharacters(String[] words, String chars) {
        int arr[] = new int[26];
        for (char c : chars.toCharArray()) {
            arr[c - 'a']++;
        }
        int ans = 0;
        for (String s : words) {
            int org[] = arr.clone();
            boolean flag = true;
            for (char c : s.toCharArray()) {
                if (org[c - 'a'] == 0) {
                    flag = false;
                    break;
                }
                org[c - 'a']--;
            }
            if (flag)
                ans += s.length();
        }
        return ans;
    }
}